From a20f9a0d590636250487d5388a4cf7df50eb0e44 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 041/296] SystemUI: Prevent NPE in Dependency.get() in
 BrightnessController

When com.android.systemui process died for some reason, the process will
be restarted and will schedule to restart services, such as
SystemUIService and KeyguardService, if start activity BrightnessDialog
before these services, then NPE will happen, the reason is that this
activity will use Dependency which is initalized in these services.

Init Dependency in BrightnessController before using it.
---
 .../SystemUI/src/com/android/systemui/settings/BrightnessController.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java b/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java
index 62eccb6c07a..039b8a2181c 100644
--- a/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java
+++ b/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java
@@ -277,6 +277,7 @@ public class BrightnessController implements ToggleSlider.Listener {
         mContext = context;
         mIcon = icon;
         mControl = control;
+        Dependency.initDependencies(context.getApplicationContext());
         mBackgroundHandler = new Handler((Looper) Dependency.get(Dependency.BG_LOOPER));
         mUserTracker = new CurrentUserTracker(mContext) {
             @Override
-- 
2.11.0

